rivertest/rivertest.go

```diff
@@ -291,22 +414,48 @@ func compareJobToInsertOpts(t testingT, jobRow *rivertype.JobRow, expectedOpts R
 		return fmt.Sprintf(" (expected job slice index %d)", index)
 	}
 
-	if expectedOpts.MaxAttempts != 0 && jobRow.MaxAttempts != expectedOpts.MaxAttempts {
-		failure(t, "Job with kind '%s'%s max attempts %d not equal to expected %d",
-			jobRow.Kind, positionStr(), jobRow.MaxAttempts, expectedOpts.MaxAttempts)
-		return false
+	var failures []string
+
+	if expectedOpts.MaxAttempts != 0 {
+		if jobRow.MaxAttempts == expectedOpts.MaxAttempts {
+			if requireNotInserted {
+				failures = append(failures, fmt.Sprintf("max attempts equal to excluded %d", expectedOpts.MaxAttempts))
+			}
+		} else {
+			if requireNotInserted {
+				return true // any one property doesn't match; assertion passes
+			} else {
+				failures = append(failures, fmt.Sprintf("max attempts %d not equal to expected %d", jobRow.MaxAttempts, expectedOpts.MaxAttempts))
+			}
+		}
```

Comment:
There's definitely a lot of duplication in these attr comparisons. Do you think it'd be worth extracting a generic helper for these comparable attrs (int, string)?

Reply:
Looked into this one. I'm not against the idea, but the helper would have to take a lot of different parameters to replicate that functionality that's in here now, and even if it's DRYer, I think overall it might obscure rather than clarify. The special cases of `ScheduledAt` and `Tags` would also be much harder to shoehorn in there and would probably have to stay on there own, thereby making the overall value a bit more dubious again.